app/components/reaction_picker/reaction_picker.js

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    Image,
+    View,
+    TouchableWithoutFeedback,
+} from 'react-native';
+
+import Emoji from 'app/components/emoji';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {hapticFeedback} from 'app/utils/general';
+import addReactionIcon from 'assets/images/icons/reaction.png';
+import {REACTION_PICKER_HEIGHT} from '../../screens/post_options/post_options_utils';
+
+export default class Reaction extends PureComponent {
+    static propTypes = {
+        addReaction: PropTypes.func.isRequired,
+        openReactionScreen: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        recentEmojis: PropTypes.array,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedIndex: -1,
+            defaultEmojis: [
+                'thumbsup',
+                'smiley',
+                'white_check_mark',
+                'heart',
+                'eyes',
+                'raised_hands',
+            ],
```

Comment:
this could be the emoji's to show, instead of having the defaultEmojis in state that won't ever change.

then you could store the emojis in state and remove line 59.

you can do this:

move defaultEmojis as a const called DEFAULT_EMOJIS;

```suggestion
            emojis: Array.from(new Set(this.props.recentEmojis.concat(DEFAULT_EMOJIS))).splice(0, 6);
```

Reply:
Awesome! and the splicing totally does away with the need to keep an index on the looping below, that's good stuff 👍